src/test/storage/dictionary_segment_test.cpp

```diff
@@ -3,68 +3,145 @@
-
-//   // Test attribute_vector size
-//   EXPECT_EQ(dict_col->size(), 6u);
-
-//   // Test dictionary size (uniqueness)
-//   EXPECT_EQ(dict_col->unique_values_count(), 4u);
-
-//   // Test sorting
-//   auto dict = dict_col->dictionary();
-//   EXPECT_EQ((*dict)[0], "Alexander");
-//   EXPECT_EQ((*dict)[1], "Bill");
-//   EXPECT_EQ((*dict)[2], "Hasso");
-//   EXPECT_EQ((*dict)[3], "Steve");
-// }
-
-// TEST_F(StorageDictionarySegmentTest, LowerUpperBound) {
-//   for (int i = 0; i <= 10; i += 2) vc_int->append(i);
-
-//   std::shared_ptr<BaseSegment> col;
-//   resolve_data_type("int", [&](auto type) {
-//     using Type = typename decltype(type)::type;
-//     col = std::make_shared<DictionarySegment<Type>>(vc_str);
-//   });
-//   auto dict_col = std::dynamic_pointer_cast<opossum::DictionarySegment<int>>(col);
-
-//   EXPECT_EQ(dict_col->lower_bound(4), (opossum::ValueID)2);
-//   EXPECT_EQ(dict_col->upper_bound(4), (opossum::ValueID)3);
-
-//   EXPECT_EQ(dict_col->lower_bound(5), (opossum::ValueID)3);
-//   EXPECT_EQ(dict_col->upper_bound(5), (opossum::ValueID)3);
-
-//   EXPECT_EQ(dict_col->lower_bound(15), opossum::INVALID_VALUE_ID);
-//   EXPECT_EQ(dict_col->upper_bound(15), opossum::INVALID_VALUE_ID);
-// }
-
-// TODO(student): You should add some more tests here (full coverage would be appreciated) and possibly in other files.
+template <typename T>
+class DictionarySegmentFixture : public DictionarySegment<T> {
+ public:
+  DictionarySegmentFixture(std::shared_ptr<opossum::ValueSegment<T>> value_segment)
+      : DictionarySegment<T>(value_segment) {}
+  int get_minimal_number_of_bits_for_dictionary_size(size_t size) {
+    return this->_get_minimal_number_of_bits_for_dictionary_size(size);
+  }
+  std::shared_ptr<BaseAttributeVector> create_attribute_vector_with_type(int minimal_bytes,
+                                                                         std::shared_ptr<ValueSegment<T>> value_segment,
+                                                                         std::shared_ptr<std::vector<T>> dictionary) {
+    return this->_create_attribute_vector_with_type(minimal_bytes, value_segment, dictionary);
+  };
+};
+
+class StorageDictionarySegmentTest : public ::testing::Test {
+ protected:
+  std::shared_ptr<opossum::ValueSegment<int>> vc_int = std::make_shared<opossum::ValueSegment<int>>();
+  std::shared_ptr<opossum::ValueSegment<std::string>> vc_str = std::make_shared<opossum::ValueSegment<std::string>>();
+};
+
+TEST_F(StorageDictionarySegmentTest, CompressSegmentString) {
+  vc_str->append("Bill");
+  vc_str->append("Steve");
+  vc_str->append("Alexander");
+  vc_str->append("Steve");
+  vc_str->append("Hasso");
+  vc_str->append("Bill");
+
+  auto dict_col = std::make_shared<DictionarySegment<std::string>>(vc_str);
+
+  // Test attribute_vector size
+  EXPECT_EQ(dict_col->size(), 6u);
+
+  // Test dictionary size (uniqueness)
+  EXPECT_EQ(dict_col->unique_values_count(), 4u);
+
+  // Test sorting
```

Comment:
One could additionally test the value access on the dictionary segment, e.g. with the` []-operator` and the `get() `accessor

Reply:
Ok, I just found the `[]-operator`, nice, but `get() `still holds I think?

Reply:
alright, `get()` gets used in a table test and thus is covered, but as you already dedicatedly test the `[]-operator` maybe you could add a test for `get()`, too